lib/mongo.ex

```diff
@@ -741,17 +747,42 @@ defmodule Mongo do
           false -> {:error, error}
         end
     end
-
   end
 
   @doc false
   @spec exec_command(GenServer.server, BSON.document, Keyword.t) :: {:ok, BSON.document | nil} | {:error, Mongo.Error.t}
   def exec_command(conn, cmd, opts) do
+    cmd
+    |> Enum.into(%{})
+    |> trace_start()
+    
     with {:ok, _cmd, {doc, event}} <- DBConnection.execute(conn, %Query{action: :command}, [cmd], defaults(opts)),
          {:ok, doc} <- check_for_error(doc, event) do
+      trace_end(event)
+
       {:ok, doc}
     end
+  end
+
+  defp trace_start(cmd) do
+    unless Mix.env() == :test do
+      measurements = %{
+        start: DateTime.utc_now |> DateTime.to_string
+      }
+      
+      :telemetry.execute([:mongo_driver, :query_start], measurements, cmd)
```

Comment:
probably it's better to split `query_start` to 2 separate parts:
`query` and `start` since there might be different mongo_driver events and it will be easier to listen only to query events:

```
:telemetry.execute([:mongo_driver, :query, :start]...
```